libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -751,6 +757,7 @@ instance (HasSpec fn a, Arbitrary (TypeSpec fn a)) => Arbitrary (Specification f
       , (10, pure baseSpec)
       ]
 
+  shrink (ExplainSpec _ x) = shrink x
```

Comment:
Here we might want to do `x : map (ExplainSpec es) shrink x` instead.

Reply:
And maybe also TrueSpec and ErrorSpec as shrinks

Reply:
done

Reply:
Oh yeah, and we want to generate these specs too!

Reply:
So I added this case to the frequency block
 , (1, ExplainSpec ["Arbitrary"] <$>  arbitrary)